src/config.ts

```diff
@@ -331,3 +276,30 @@ export const config = configOrder({
     },
   },
 })
+
+// finds the default shell start commmand
+export async function setAutoShell(): Promise<void> {
+  let shellStartCommand
+  if (process.platform === "win32") {
+    // Windows
+    try {
+      shellStartCommand = await which("pwsh.exe")
+    } catch (e1) {
+      try {
+        shellStartCommand = await which("powershell.exe")
+      } catch (e2) {
+        // keep default
+      }
+    }
+  }
+
+  if (shellStartCommand && atom.config.get("terminal.shell") === getDefaultShell()) {
+    atom.config.set("terminal.shell", shellStartCommand)
+  }
+}
+
+// set shell command automatically on first install
+if (localStorage.getItem("terminal.autoShellSet") === null) {
+  localStorage.setItem("terminal.autoShellSet", "true")
+  setAutoShell()
+}
```

Comment:
This is all we need to set the shell on install.

Reply:
If we want to set this only once, we don't need to make this async. We will not even need to use localStorage.

Reply:
What is the broken behavior? Don't we just want to default to powershell if it is available?

We still want it to be async so it doesn't block the main thread.

Reply:
Local storage is just used so this is only called on install and never again

Reply:
If the user changes the default shell I think they are competent enough to know how to change it back if they want it.

Reply:
> If the user changes the default shell I think they are competent enough to know how to change it back if they want it.

That's a good assumption for now, but we need to fix it. 

> What is the broken behavior? Don't we just want to default to powershell if it is available?

I want this option to be similar to all other configuration defaults. As I said, we probably should fix this inside Atom, or add a "reset cache" button.

Reply:
> we probably should fix this inside Atom
Yes but until that happens I think this is a good enough work around. It solves the problem with minimal code and not blocking the main thread at all.